redbot/cogs/mod/kickban.py

```diff
@@ -173,6 +173,18 @@ class KickBanMixin(MixinMeta):
             await ctx.send(_("I cannot do that due to discord hierarchy rules"))
             return
         audit_reason = get_audit_reason(author, reason)
+        if reason is None:
+            reason = "No reason was given."
+        self == self.bot.get_cog("Mod")
+        toggle = await self.settings.guild(guild).toggle_dm()
+        if toggle:
+            if guild.me.top_role >= user.top_role and user != guild.owner and author != user:
```

Comment:
One can only kick users who's highest role is lower than theirs - you should probably use `>` instead of `>=` there, having the same top role with someone prevents them from getting kicked / banned.